src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,68 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { FilterStatus } from '../../types/FilterStatus';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Filter = {
+  filterStatus: string;
+  onFilter: (status: FilterStatus) => void;
+  query: string;
+  onChangeQuery: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onClearSearch: () => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Filter> = ({
+  onFilter,
```

Comment:
```suggestion
  onChangeFilter,
```

will be more informative